components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthenticatedSubjectIdentifierHandler.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.handler.request.impl;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.AuthenticatorConfig;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.SequenceConfig;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.handler.request.AbstractPostAuthnHandler;
+import org.wso2.carbon.identity.application.authentication.framework.handler.request.PostAuthnHandlerFlowStatus;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.List;
+
+import static org.wso2.carbon.identity.application.authentication.framework.handler.request.PostAuthnHandlerFlowStatus.UNSUCCESS_COMPLETED;
+
+/**
+ * This PostAuthenticationHandler is responsible for setting subject identifier related with authenticated user.
+ */
+public class PostAuthenticatedSubjectIdentifierHandler extends AbstractPostAuthnHandler {
+
+    private static final Log log = LogFactory.getLog(PostAuthenticatedSubjectIdentifierHandler.class);
+    private static PostAuthenticatedSubjectIdentifierHandler instance;
+
+    /**
+     * To get the singleton instance of post authentication handler.
+     *
+     * @return instance of {@link PostAuthenticatedSubjectIdentifierHandler}.
+     */
+    public static PostAuthenticatedSubjectIdentifierHandler getInstance() {
+
+        if (instance == null) {
+            synchronized (PostAuthenticatedSubjectIdentifierHandler.class) {
+                if (instance == null) {
+                    instance = new PostAuthenticatedSubjectIdentifierHandler();
+                }
+            }
+        }
+        return instance;
+    }
+
+    @Override
+    public int getPriority() {
+        /*
+        The priority of this authentication handler should be greater than JIT and association post authentication
+        handler as subject identifier need to be set after all these handling.
+         */
+        return 22;
```

Comment:
Change this to get the configuration from identity.xml. If not exist, take it from here